engine/lib/concrete_ident/concrete_ident.ml

```diff
@@ -635,6 +634,14 @@ module Create = struct
   let constructor name =
     let path = name.def_id.path @ [ { data = Ctor; disambiguator = 0 } ] in
     { name with def_id = { name.def_id with path } }
+
+  let add_disambiguator name disambiguator =
+    let path = name.def_id.path in
+    let last = List.last_exn path in
+    let path =
+      List.drop_last_exn path @ [ { data = last.data; disambiguator } ]
```

Comment:
Shopuld we be using an "_exn" function here? If needed, let's put a comment saying why this cannot fail.

Reply:
Done